src/hardware/link-mso19/protocol.c

```diff
@@ -199,7 +199,7 @@ SR_PRIV int mso_parse_serial(const char *iSerial, const char *iProduct,
 		u3 = 500;
 	devc->vbit = ((double)u1) / 10000.0 / 1000.0;
 	devc->dac_offset = u2;
-	devc->offset_range = u3;
+	devc->offset_vbit = 3.0 / u3;
```

Comment:
Initially I had missed this "3.0" number - where is that from ?

Reply:
I'm not exactly sure exactly. It is basically taken from this other project https://github.com/tkrmnz/mso19fcgi/blob/master/main.c#L332

It's been a while since I last looked at that code, so it is a bit fuzzy, but I think the off by factor of 1000 is probably due to mV to V conversion somewhere differing in their code vs ours.

I'm pretty sure I tested it and it worked a long time ago.

Reply:
I think that's probably the last "magic number" that would be nice to get rid of. TBH I haven't gone back to look at what `offset_vbit` actually does, so maybe this is a non-issue.

Reply:
@knarfS The existing code for mso-19 is riddled with this type of calculation. I tried to backward engineer and make clear everything I could... If having 100% knowledge of the protocol is a requirement for you to merge this code, then I am going to give up and abandon all hope.

If you are willing to budge a little on your perfect ideals to let this one slide, then I can fix up all your other comments and hopefully get this hardware working for other folks who would like to use it. I can at least make it an aptly named macro if that helps.

It would be sad after putting all the effort I have into this code to just throw it all away.

Reply:
> > It is basically taken from this other project https://github.com/tkrmnz/mso19fcgi/blob/master/main.c#L332
>
> That makes it even worse: I don't know how much code is copied from there

I don't know about any of the original code, but **none of the code I wrote is copied**. Yes, some of the calculations are based on the other project because I have no idea where else to get those types of calculations... Besides, that project is already linked to in the mso-19 wiki (which is how I found it in the first place).

Reply:
I'm totally aware that you didn't copied that code, but when there is code in here, that was taken from that other project, then there should be a note with a link at the top of this file and probably also in the commit message. A link in the wiki is obviously not enough.